code/__DEFINES/traits.dm

```diff
@@ -19,6 +19,9 @@
 				SEND_SIGNAL(target, SIGNAL_ADDTRAIT(trait), trait); \
 			} \
 		} \
+		if (duration) { \
+			addtimer(CALLBACK(target, .proc/REMOVE_TRAIT, target, trait, source), duration, TIMER_UNIQUE|TIMER_OVERRIDE); \
```

Comment:
This isn't even going to work.

Reply:
> Please stop making untested web edits

This is a WIP that I was about to draft before you closed it.  The reason for the PRs creation was to get feedback.

> also please don't add this without any consumers

This would be used for the memory feature I'm in the process of building.  It's useful enough to where I think it should be it's own PR, since it alters the functionality of a already existing subsystem.